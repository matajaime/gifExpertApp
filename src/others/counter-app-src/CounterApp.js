import React, { useState } from 'react';
import PropTypes from 'prop-types';


const CounterApp = ({ value = 5 }) => {
    const [contador, setContador] = useState(value);

    const handleAdd = ( event ) => {
        //setContador( contador + 1 );
        setContador( (c) => c + 1 );
    };
    const handleSub = ( event ) => {
        setContador( (c) => c - 1 );
    };
    const handleReset = ( event ) => {
        setContador( value );
    };
    //console.log("Me llamaron");
    return (
        <>
        <h1>CounterApp</h1>
        <h2>{ contador }</h2>
        <button onClick={ handleAdd }>+1</button>
        <button onClick={ handleReset } id="btnReset">Reset</button>
        <button onClick={ handleSub }>-1</button>
        </>
    );
}

CounterApp.propTypes = {
    value: PropTypes.number
}

export default CounterApp;
