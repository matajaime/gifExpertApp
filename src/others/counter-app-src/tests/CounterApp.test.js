import React from 'react';
import { shallow } from 'enzyme';
import CounterApp from '../CounterApp';

describe('Pruebas CounterApp',()=>{
    const wrapper = shallow(<CounterApp />);
    test('Prueba de snapShot base sin parametros', ()=>{
        expect(wrapper).toMatchSnapshot();        
    });
    test('Verificar contador inicial enviado por parámetro',()=>{
        const contadorInicial = 100;
        const wrapper = shallow(<CounterApp value={contadorInicial} />);
        const contador = Number(wrapper.find('h2').text());
        expect( contador ).toBe( contadorInicial );
    });
    test('Verificar incremento de contador',()=>{
        const buttonAdd = wrapper.find('button').at(0);
        const contadorOld = Number(wrapper.find('h2').text());
        buttonAdd.simulate('click');
        const contadorNew = Number(wrapper.find('h2').text());
        expect( contadorNew ).toBe( contadorOld + 1);
    });
    test('Verificar reducción de contador',()=>{
        const buttonSub = wrapper.find('button').at(2);
        const contadorOld = Number(wrapper.find('h2').text());
        buttonSub.simulate('click');
        const contadorNew = Number(wrapper.find('h2').text());
        expect( contadorNew ).toBe( contadorOld - 1);
    });
    test('Verificar reset',()=>{
        const contadorInicial = 111;
        const wrapper = shallow(<CounterApp value={contadorInicial} />);
        const buttonSub = wrapper.find('button').at(2);
        buttonSub.simulate('click');
        const buttonReset = wrapper.find('#btnReset');
        buttonReset.simulate('click');
        const contador = Number(wrapper.find('h2').text());
        expect( contador ).toBe( contadorInicial );
    });
});