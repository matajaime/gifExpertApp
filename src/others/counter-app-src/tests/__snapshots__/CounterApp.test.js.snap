// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas CounterApp Prueba de snapShot base sin parametros 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    5
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    id="btnReset"
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
